core/src/trezor/ui/layouts/tt_v2/fido.py

```diff
@@ -0,0 +1,101 @@
+from typing import TYPE_CHECKING
+
+from trezor.enums import ButtonRequestType
+
+import trezorui2
+
+from ..common import interact
+from . import _RustLayout
+
+if TYPE_CHECKING:
+    from trezor.loop import AwaitableTask
+    from trezor.wire import GenericContext
+
+
+if __debug__:
+    from trezor import io
+    from ... import Result
+
+    class _RustFidoLayoutImpl(_RustLayout):
+        def create_tasks(self) -> tuple[AwaitableTask, ...]:
+            return (
+                self.handle_timers(),
+                self.handle_input_and_rendering(),
+                self.handle_swipe(),
+                self.handle_debug_confirm(),
+            )
+
+        async def handle_debug_confirm(self) -> None:
+            from apps.debug import confirm_signal
+
+            try:
+                await confirm_signal()
+            except Result as r:
+                if r.value is not trezorui2.CONFIRMED:
+                    raise
+            else:
+                return
+
+            for event, x, y in (
+                (io.TOUCH_START, 220, 220),
+                (io.TOUCH_END, 220, 220),
+            ):
+                msg = self.layout.touch_event(event, x, y)
+                self.layout.paint()
+                if msg is not None:
+                    raise Result(msg)
+
+    _RustFidoLayout = _RustFidoLayoutImpl
+
+else:
+    _RustFidoLayout = _RustLayout
+
+
+async def confirm_fido(
+    ctx: GenericContext | None,
+    header: str,
+    app_name: str,
+    icon_name: str | None,
+    accounts: list[str | None],
+) -> int:
+    """Webauthn confirmation for one or more credentials."""
+    # `trezorui2.confirm_webauthn` is expecting a list of tuples
+    # (app_name, account_name) to allow for the `pageable` situation.
```

Comment:
This comment is outdated.

Reply:
[e5359f4](https://github.com/trezor/trezor-firmware/pull/2556/commits/e5359f41f25496e1e6b1876a07eb4b39c2ea23f1)